registry/storage/driver/s3/s3.go

```diff
@@ -314,6 +321,9 @@ func (d *driver) WriteStream(path string, offset int64, reader io.Reader) (total
 				}
 			}
 		}
+
+		d.putbuf(buf)
```

Comment:
This doesn't match up with the existing logic, currently this block never reallocs buf